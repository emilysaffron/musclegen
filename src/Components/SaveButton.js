import styled from "@emotion/styled";
import ToggleModal from "../Helpers/ToggleModal";
import WriteUserData from "../Helpers/WriteUserData";
import { Link } from "@reach/router";
const StyledButton = styled.button`
  border-radius: 20px;
  width: 150px;
  height: 90px;
  padding: 20px;
  margin: 10px;
  font-family: Palatino, URW Palladio L, serif;
  color: black;
  &:hover {
    cursor: pointer;
  }
  outline: none;
`;

const ButtonsContainer = styled.div`
  display: flex;
`;

const SaveButton = ({ modal, toggleModal, currentPlan }) => {
  const handleCancelClick = () => {
    ToggleModal(toggleModal, modal);
  };
  const handleConfirmClick = () => {
    WriteUserData(currentPlan);
    ToggleModal(toggleModal, modal);
  };

  return (
    <Link to={"/"}>
      <ButtonsContainer>
        <StyledButton onClick={() => handleConfirmClick()}>
          Save Workout
        </StyledButton>
        <StyledButton onClick={() => handleCancelClick()}>
          Don't Save
        </StyledButton>
      </ButtonsContainer>
    </Link>
  );
};

export default SaveButton;
