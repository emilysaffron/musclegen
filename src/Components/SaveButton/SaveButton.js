import styled from "@emotion/styled";
import WriteUserData from "../../Helpers/WriteUserData";
import { Link } from "@reach/router";
const StyledButton = styled.button`
  border-radius: 20px;
  width: 150px;
  height: 90px;
  padding: 20px;
  margin: 10px;
  cursor: pointer;
  outline: none;
`;

const ButtonsContainer = styled.div`
  display: flex;
`;

const SaveButton = ({ modal, toggleModal, currentPlan }) => {
  const handleCancelClick = () => {
    toggleModal(!modal);
  };
  const handleConfirmClick = () => {
    WriteUserData(currentPlan);
    toggleModal(modal);
  };

  return (
    <Link to={"/"}>
      <ButtonsContainer>
        <StyledButton onClick={() => handleConfirmClick()}>
          Save Workout
        </StyledButton>
        <StyledButton onClick={() => handleCancelClick()}>
          Don't Save
        </StyledButton>
      </ButtonsContainer>
    </Link>
  );
};

export default SaveButton;
