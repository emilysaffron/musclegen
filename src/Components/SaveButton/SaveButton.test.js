import renderer from "react-test-renderer";
import SaveButton from "./SaveButton";

it("SaveButton Snapshot", () => {
  const tree = renderer
    .create(
      <SaveButton modal={true} toggleModal="toggleModal" currentPlan={[]} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
