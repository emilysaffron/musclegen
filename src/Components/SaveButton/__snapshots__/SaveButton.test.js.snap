// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButton Snapshot 1`] = `
<a
  aria-current="page"
  href="/"
  onClick={[Function]}
>
  <div
    className="css-ov1ktg"
  >
    <button
      className="css-ljtnwa"
      onClick={[Function]}
    >
      Save Workout
    </button>
    <button
      className="css-ljtnwa"
      onClick={[Function]}
    >
      Don't Save
    </button>
  </div>
</a>
`;
